File: src/main/java/com/pauljoda/modularsystems/storage/gui/GuiStorageHopping.java
package com.pauljoda.modularsystems.storage.gui;

import com.pauljoda.modularsystems.storage.tiles.TileStorageHopping;
import com.teambr.bookshelf.client.gui.GuiBase;
import com.teambr.bookshelf.client.gui.component.control.GuiComponentSlider;
import com.teambr.bookshelf.inventory.ContainerGeneric;
import com.teambr.bookshelf.manager.PacketManager;

/**
 * Modular-Systems
 * Created by Paul Davis on 7/27/2015
 */
public class GuiStorageHopping extends GuiBase<ContainerGeneric> {
    protected TileStorageHopping hopper;
    /**
     * Constructor for All Guis
     */
    public GuiStorageHopping(TileStorageHopping tile) {
        super(new ContainerGeneric(), 100, 50, "inventory.storageHopping.title");
        hopper = tile;
        addComponents();
    }

    @Override
    public void addComponents() {
        if(hopper != null) {
            components.add(new GuiComponentSlider<Integer>(10, 25, 80, GuiComponentSlider.generateNumberList(0, 10), hopper.range) {
                @Override
                public void onValueChanged(Integer value) {
                    hopper.range = value;
                    PacketManager.updateTileWithClientInfo(hopper);
                }
            });
        }
    }
}

File: src/main/scala/com/teambr/modularsystems/storage/gui/GuiStorageHopping.scala
package com.teambr.modularsystems.storage.gui

import com.teambr.bookshelf.client.gui.GuiBase
import com.teambr.bookshelf.client.gui.component.control.GuiComponentSlider
import com.teambr.bookshelf.common.container.ContainerGeneric
import com.teambr.bookshelf.network.PacketManager
import com.teambr.modularsystems.storage.tiles.TileStorageHopping

/**
 * This file was created for Modular-Systems
 *
 * Modular-Systems is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis <pauljoda>
 * @since August 10, 2015
 */
class GuiStorageHopping(val tile: TileStorageHopping) extends GuiBase[ContainerGeneric](new ContainerGeneric, 100, 50, "inventory.storageHopping.title") {
    override def addComponents() : Unit = {
       components += new GuiComponentSlider[Int](10, 25, 80, List.range(0, 11), tile.range) {
           override def onValueChanged(value : Int): Unit = {
               tile.range = value
               PacketManager.updateTileWithClientInfo(tile)
           }
       }
    }
}
